### PR TITLE
add cross accounts and region testing in LS docs

### PR DESCRIPTION
**Pull request by sannya-singal**
Localstack now supports multiple accounts and regions compatibility. This document guides the contributors to test their changes against cross accounts and region.

### CODE REVIEW

File: content/en/contributing/multi-account-region-testing.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Multiple account and region Testing"
+weight: 7
+description: >
+  How to test multi-account and multi-region compatibility.
+aliases:
+  - /developer-guide/multi-account-region-testing/
```
Comment (HarshCasper) [inline content/en/contributing/multi-account-region-testing.md:7]: Don't need to add alias

File: content/en/contributing/multi-account-region-testing.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Multiple account and region Testing"
```
Comment (HarshCasper) [inline content/en/contributing/multi-account-region-testing.md:2]:
```suggestion
title: "Multi account and region testing"
```

File: content/en/contributing/multi-account-region-testing.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+weight: 7
+description: >
+  How to test multi-account and multi-region compatibility.
+aliases:
+  - /developer-guide/multi-account-region-testing/
+---
+
+LocalStack community supports multiple accounts and regions compatibility. This document describes how to test your changes against cross accounts and regions.
```
Comment (HarshCasper) [inline content/en/contributing/multi-account-region-testing.md:10]:
```suggestion
LocalStack supports multi accounts and regions compatibility. This document describes how to test your changes against cross accounts and regions.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+## Test changes locally with non-default credentials
+
+In order to test your changes on your machine for multi-accounts and region compatibility, i.e. for non-default values other than `000000000000` for account ID or `us-east-1` for region, set the following environment variables to any random non-default values, for example:  
+
+- `TEST_AWS_ACCOUNT_ID=111111111111`
+- `TEST_AWS_ACCESS_KEY_ID=111111111111`
+- `TEST_AWS_REGION=us-west-1`
+- `TEST_AWS_SECRET_ACCESS_KEY=test1`
+
+We can additionally prefer to create a commit eg: [da3f8d5](https://github.com/localstack/localstack/pull/9751/commits/da3f8d5f2328adb7c5c025722994fea4433c08ba) to test the pipeline for non-default credentials against your changes.
```
Comment (HarshCasper) [inline content/en/contributing/multi-account-region-testing.md:31]:
```suggestion
You can additionally prefer to create a commit eg: [da3f8d5](https://github.com/localstack/localstack/pull/9751/commits/da3f8d5f2328adb7c5c025722994fea4433c08ba) to test the pipeline for non-default credentials against your changes.
```

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: content/en/contributing/multi-account-region-testing.md (hunk 3 above)
Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:10]: Since only community repo is multi accounts compatible for now, does it make sense to explicitly highlight this here @HarshCasper?

Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:10]: like localstack community instead of just localstack? 🤔

Comment (HarshCasper) [reply content/en/contributing/multi-account-region-testing.md]: We wish to remove any pricing info, and just mention `LocalStack` as a product in itself. Hence we can remove the `community` keyword.

**Review by alexrashed — CHANGES_REQUESTED** [review]
I think we should not advise our contributors to change the pipeline code, as this seems a bit unsafe and will trigger some unwanted codeowner-triggered reviewers.
If it is necessary for community contributors to trigger the pipeline, we should make it easier. However, for now I think it should be enough that we start a run of the pipeline if we see that a contribution might break something?

This is why I would suggest to have this page describe - in case you have the permissions - how to trigger the pipeline or how to test it locally.

**Review by alexrashed — COMMENTED** [review]
This is going into a great direction! I have a few nitpicks, and some suggestions (most importantly if we should add some very basic guidelines on how to make inter-service requests to avoid breaking it in the first place). Afterwards this is looking good from my side! It will be a great reference to be used in Community PRs, but also for newcomers to the team! 🚀 
Maybe @viren-nadkarni also has some ideas / an opinion on this?

File: content/en/contributing/multi-account-region-testing.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross accounts against the changes. To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credenitals every night at 1:00am UTC.
+
+## Manually trigger the workflow
+
+To manually trigger the scheduled workflow in case you have the permissions, you can change set the value of `randomize-aws-credentials` to `true` in the [workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml#L13). This will trigger the tests with randomised account and region credentials.
+
+{{< alert title="Note">}}
+Make sure to change the value back to `false` after the tests are completed. 
+{{< /alert >}}
```
[the remark this answers is not in the document]
Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:18]: This is in general for anyone wants to create a commit similar to one here- [10478](https://github.com/localstack/localstack/pull/10478/commits/b455596ee60ed012a3be8c5ec94ce4f5823bd4a7). This is to trigger the pipeline to check the results for non-defaults creds when someone doesnt has access to webapp directly.

File: content/en/contributing/multi-account-region-testing.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
+title: "Multi account and region testing"
```
[the remark this answers is not in the document]
Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:2]: Got a review to change this [here](https://github.com/localstack/docs/pull/1129#discussion_r1528032339) earlier. @HarshCasper any suggestions here?

File: content/en/contributing/multi-account-region-testing.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexrashed) [reply content/en/contributing/multi-account-region-testing.md:18]: Ah, right, I see. And in fact, I overlooked the actual issue here:
What I meant in a previous comment to change this section describing how to manually trigger the pipeline with the parameter enabled _in the UI of CircleCI_, not in the code.
When the section above does not describe a code change anymore, this note also becomes unnecessary.

File: content/en/contributing/multi-account-region-testing.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+weight: 7
```
[the remark this answers is not in the document]
Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:3]: I discussed this with @HarshCasper since I wasn't sure about this. @alexrashed do you think it should be 8 here?

[the remark this answers is not in the document]
Comment (alexrashed) [reply content/en/contributing/multi-account-region-testing.md:3]: I don't have a strong opinion here, but I would say the parity testing guide is a bit more important (because basically _every_ PR needs to use it, while this guide is only interesting for code touching the inter-service-communication).
The parity guide has weight 7, the terraform testing has 10, so I think 8 would work well.
But again, no strong opinion :)

File: content/en/contributing/multi-account-region-testing.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+description: >
+  How to test multi-account and multi-region compatibility.
+---
+
+LocalStack supports multi accounts and regions compatibility. This document describes how to test your changes against cross accounts and regions.
```
Comment (viren-nadkarni) [inline content/en/contributing/multi-account-region-testing.md:8]: Please prefer having one sentence on each line, this helps with diffs in future changes.

File: content/en/contributing/multi-account-region-testing.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross accounts against the changes. To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credenitals every night at 1:00am UTC.
```
Comment (viren-nadkarni) [inline content/en/contributing/multi-account-region-testing.md:10]:
```suggestion
We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross accounts against the changes. To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credentials every night at 1:00am UTC.
```

File: content/en/contributing/multi-account-region-testing.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sannya-singal) [reply content/en/contributing/multi-account-region-testing.md:18]: So to conclude, we want this part to include the param enabling through the webapp and not the code. Right @alexrashed?

File: content/en/contributing/multi-account-region-testing.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Multi-account and Multi-region Testing"
+weight: 8
+description: >
+  How to test multi-account and multi-region compatibility.
+---
+
+LocalStack supports multi accounts and regions compatibility.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:8]: nit: I think we should stick to the terms combined with a dash: `multi-account`, `multi-region`, `cross-account`, `inter-service-communication`. What's your opinion? Are there any guidelines or best practices on this, @HarshCasper?
```suggestion
LocalStack supports multi-accounts and multi-regions compatibility.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Multi-account and Multi-region Testing"
+weight: 8
+description: >
+  How to test multi-account and multi-region compatibility.
+---
+
+LocalStack supports multi accounts and regions compatibility. 
+This document describes how to test your changes against cross accounts and regions.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:9]:
```suggestion
This document describes how to test your changes for compatibility with cross-account and cross-region requests.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross accounts against the changes. To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credentials every night at 1:00am UTC.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:11]: I think this isn't really useful information here. Maybe we could move this down to the section on manually triggering the pipeline? Maybe rephrase the section below to "Test changes in CI with non-default credentials" (in contrast and right next to "Test changes locally with non-default credentials")?
Also this line contains multiple sentences (we try to have a single sentence each line, see best practices in the README).

File: content/en/contributing/multi-account-region-testing.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+
+## Tips to make your changes compatible with multi-accounts and regions
+
+In order to make your changes compatible with multi-accounts and regions, you should follow the below tips:
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:16]:
```suggestion
In order to make your changes compatible with multi-accounts and multi-regions, you should follow the below tips:
```

File: content/en/contributing/multi-account-region-testing.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+- For cross accounts, in inter service communication for many integrations, you can specify a role, with which permissions the source service makes a request to the target service, to access other services resource. This role should be in source account. When writing an AWS validated test case, you need to properly configure IAM roles.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:18]:
```suggestion
- For cross-accounts, in inter-service-communication for many integrations, you can specify a role, with which permissions the source service makes a request to the target service, to access another service's resource.
  This role should be in source account.
  When writing an AWS validated test case, you need to properly configure IAM roles.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+  How to test multi-account and multi-region compatibility.
+---
+
+LocalStack supports multi accounts and regions compatibility. 
+This document describes how to test your changes against cross accounts and regions.
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross accounts against the changes. To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credentials every night at 1:00am UTC.
+
+
+## Tips to make your changes compatible with multi-accounts and regions
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:14]:
```suggestion
## Tips to make your changes compatible with multi-accounts and multi-regions
```

File: content/en/contributing/multi-account-region-testing.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+## Test changes locally with non-default credentials
+
+In order to test your changes on your machine for multi-accounts and region compatibility in case you have the permissions, i.e. for non-default values other than `000000000000` for account ID or `us-east-1` for region, set the following environment variables to any random non-default values, for example:  
+
+- `TEST_AWS_ACCOUNT_ID=111111111111`
+- `TEST_AWS_ACCESS_KEY_ID=111111111111`
+- `TEST_AWS_REGION=us-west-1`
+- `TEST_AWS_SECRET_ACCESS_KEY=test1`
+
+You can additionally prefer to create a commit eg: [da3f8d5](https://github.com/localstack/localstack/pull/9751/commits/da3f8d5f2328adb7c5c025722994fea4433c08ba) to test the pipeline for non-default credentials against your changes. Note that within the tests we must use `account_id`, `secondary_account_id`, `region_name`, `secondary_region_name` fixtures and avoid importing `localstack.constants.TEST_*` values.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:69]:
```suggestion
You can additionally prefer to create a commit eg: [da3f8d5](https://github.com/localstack/localstack/pull/9751/commits/da3f8d5f2328adb7c5c025722994fea4433c08ba) to test the pipeline for non-default credentials against your changes.
Note that within the tests we must use `account_id`, `secondary_account_id`, `region_name`, `secondary_region_name` fixtures and avoid importing `localstack.constants.TEST_*` values. 
```

File: content/en/contributing/multi-account-region-testing.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+- You should always refer the official AWS API docs and check how the the application is communicating to each other. 
+    
+    For example: 
+    Firehose to s3 refer to [Offical AWS Docs](https://docs.aws.amazon.com/firehose/latest/dev/controlling-access.html#cross-account-delivery-s3).
+
+
+## Manually trigger the workflow
+
+To manually trigger the scheduled workflow in case you have the permissions, you can change set the value of `randomize-aws-credentials` to `true` in the [workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml#L13). This will trigger the tests with randomised account and region credentials.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:58]: Please describe here how you can trigger the workflow with the proper parameters in the UI, and not with a commit.
Also, this line contains multiple sentences.

**Review by alexrashed — CHANGES_REQUESTED** [review]

File: content/en/contributing/multi-account-region-testing.md [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+
+LocalStack supports multi-account and multi-region compatibility. 
+This document describes how to test your changes for compatibility with cross-account and cross-region requests.
+
+## Tips to make your changes compatible with multi-account and multi-region
+
+In order to make your changes compatible with multi-account and multi-region, you should follow the below tips:
+
+- For cross-account, in inter-service-communication for many integrations, you can specify a role, with which permissions the source service makes a request to the target service, to access another service's resource. 
+This role should be in source account.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:16]: nit:
```suggestion
This role should be in the source account. 
```

File: content/en/contributing/multi-account-region-testing.md [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+## Tips to make your changes compatible with multi-account and multi-region
+
+In order to make your changes compatible with multi-account and multi-region, you should follow the below tips:
+
+- For cross-account, in inter-service-communication for many integrations, you can specify a role, with which permissions the source service makes a request to the target service, to access another service's resource. 
+This role should be in source account. 
+When writing an AWS validated test case, you need to properly configure IAM roles.
+
+    For example: 
+    In the test case `test_api_gateway_kinesis_integration` we have specified a [role](https://github.com/localstack/localstack/blob/ae31f63bb6d8254edc0c85a66e3c36cd0c7dc7b0/tests/aws/services/apigateway/test_apigateway_basic.py#L2017-L2022) in the apigateway which has permissions to access the target kinesis account.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:20]:
```suggestion
    In the test case [`test_api_gateway_kinesis_integration`](https://github.com/localstack/localstack/blob/0c2d8a103a9a06ade4ef1c18eacf5888d2b8adcf/tests/aws/services/apigateway/test_apigateway_basic.py#L2047) we have specified a [role](https://github.com/localstack/localstack/blob/ae31f63bb6d8254edc0c85a66e3c36cd0c7dc7b0/tests/aws/services/apigateway/test_apigateway_basic.py#L2017-L2022) which has permissions to access the target kinesis account.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+    For example: 
+    Firehose to s3 refer to [Offical AWS Docs](https://docs.aws.amazon.com/firehose/latest/dev/controlling-access.html#cross-account-delivery-s3).
+
+
+## Test changes in CI with non-default credentials
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross-account against the changes. 
+To achieve that, we have a [scheduled workflow](https://github.com/localstack/localstack/blob/master/.circleci/config.yml) on [LocalStack](https://github.com/localstack/localstack), which executes the tests with randomised account and region credentials every night at 1:00am UTC.
+
+To manually trigger the scheduled workflow through circleci UI(in case you have the permissions), in order to test your changes with randomised account and region credentials, you can perform the following steps:
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:60]:
```suggestion
To manually trigger the workflow through the webinterface of CircleCI (in case you have the permissions), in order to test your changes with randomized account and region credentials, you can perform the following steps: 
```

File: content/en/contributing/multi-account-region-testing.md [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+
+To manually trigger the scheduled workflow through circleci UI(in case you have the permissions), in order to test your changes with randomised account and region credentials, you can perform the following steps: 
+- Go to the [localStack](https://app.circleci.com/pipelines/github/localstack/localstack) project repository on circleci UI.
+- Select a branch for which you want to trigger the scheduled workflow from the filters section.
+- Now click on the `Trigger pipeline` button on the right and add the following variables:
+    1. Parameter type to `string`
+    2. Name to `randomize-aws-credentials`
+    3. Value to `true`
+
+and press the `Trigger pipeline` button to trigger the workflow.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:68]:
```suggestion
- Go to the [LocalStack](https://app.circleci.com/pipelines/github/localstack/localstack) project repository on CircleCI.
- Select a branch for which you want to trigger the workflow from the filters section.
- Now click on the `Trigger pipeline` button on the right and add the following variables:
    1. Parameter type to `string`
    2. Name to `randomize-aws-credentials`
    3. Value to `true`
- Press the `Trigger pipeline` button to trigger the workflow.
```

File: content/en/contributing/multi-account-region-testing.md [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+
+- You should always refer the official AWS API docs and check how the the application is communicating to each other. 
+    
+    For example: 
+    Firehose to s3 refer to [Offical AWS Docs](https://docs.aws.amazon.com/firehose/latest/dev/controlling-access.html#cross-account-delivery-s3).
+
+
+## Test changes in CI with non-default credentials
+
+We regularly run the circleci test jobs of the LocalStack community repository to check the compatibility of cross-account against the changes.
```
Comment (alexrashed) [inline content/en/contributing/multi-account-region-testing.md:57]: There are several references to "Community" still in this page, but @HarshCasper mentioned in a comment above that we should get rid of these.
```suggestion
We regularly run the circleci test jobs of the LocalStack repository to check the compatibility of cross-account against the changes. 
```

**Review by alexrashed — CHANGES_REQUESTED** [review]
Thanks for applying all the comments! I think this should be the last revision, basically only changing a bit on the formatting and some typos. 🚀 
Btw.: You can also just bulk-apply all suggestions in a single commit with the "Add suggestion to batch" feature of GitHub.

**Review by alexrashed — APPROVED** [review]
Awesome! Thanks a lot for addressing all the comments! This page will be super-helpful in guiding contributors towards writing multi-accounts and multi-regions compatible code! 🚀 📝

**Comment by sannya-singal** [comment]
Thanks Alex for your guidance and reviews 🎉 🙏

**Comment by sannya-singal** [comment]
Thanks @alexrashed for your guidance and reviews 🎉 🙏
